examples/stream-collector/src/rando.cc

```diff
@@ -0,0 +1,250 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+Copyright 2016 Intel Corporation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "rando.h"
+
+#include <map>
+#include <string>
+#include <time.h>
+#include <vector>
+#include <iostream>
+#include <future>
+#include <thread>
+#include <chrono>
+
+#include <snap/config.h>
+#include <snap/plugin.h>
+#include <snap/metric.h>
+#include <snap/flags.h>
+#include <snap/proxy/stream_collector_proxy.h>
+
+using Plugin::Config;
+using Plugin::ConfigPolicy;
+using Plugin::Metric;
+using Plugin::Meta;
+using Plugin::Type;
+using Plugin::Flags;
+using Plugin::RpcType;
+//using Plugin::StreamChannel;
+
+using std::cout;
+using std::endl;
+
+enum supported_types
+{
+    float32,
+    float64,
+    int32,
+    int64,
+    uint32,
+    uint64,
+    boolean,
+    string
+};
+
+std::map<std::string, int> metric_types{
+    {"float32", supported_types::float32},
+    {"float64", supported_types::float64},
+    {"int32", supported_types::int32},
+    {"int64", supported_types::int64},
+    {"uint32", supported_types::uint32},
+    {"uint64", supported_types::uint64},
+    {"boolean", supported_types::boolean},
+    {"string", supported_types::string}
+};
+
+const ConfigPolicy Rando::get_config_policy() {
+    ConfigPolicy policy;
+    policy.add_rule({"intel", "cpp"},
+    Plugin::StringRule{
+        "username",
+        {"root", false}
+    });
+    policy.add_rule({"intel", "cpp", "mock", "randomnumber", "float32"},
+    Plugin::StringRule{
+        "password",
+        {"h4ck3r", true}
+    });
+    return policy;
+}
+
+std::vector<Metric> Rando::get_metric_types(Config cfg) {
+    std::vector<Metric> metrics = {
```

Comment:
I suggest creating metrics using recently added method, see [it](https://github.com/intelsdi-x/snap-plugin-lib-cpp/blob/430df640021af5e0cc1bf237bc6fe5016b756477/examples/collector/src/rando.cc#L84)